RecoParticleFlow/PFClusterProducer/src/PFClusterEMEnergyCorrector.cc

```diff
@@ -168,104 +92,165 @@ void PFClusterEMEnergyCorrector::correctEnergies(const edm::Event &evt, const ed
     reco::PFCluster &cluster = cs[idx];
     
     double e = cluster.energy();
-    double pt = cluster.pt(); 
-    
+    auto const& crep = cluster.position();
+    double pt = cluster.energy()*crep.rho()/crep.r();
+   
     //limit raw energy value used to evaluate corrections
     //to avoid bad extrapolation
     double evale = e;
-    if (_maxPtForMVAEvaluation>0. && pt>_maxPtForMVAEvaluation) {
-      evale *= _maxPtForMVAEvaluation/pt; 
+    if (maxPtForMVAEvaluation_ > 0. && pt > maxPtForMVAEvaluation_) {
+      evale *= maxPtForMVAEvaluation_/pt; 
     }
     
     double invE = (e == 0.) ? 0. : 1./e; //guard against dividing by 0.
     
-    int size = lazyTool.n5x5(cluster);
+    bool iseb = (cluster.layer() == PFLayer::ECAL_BARREL);
+    int clusFlag = 0;
     
-    bool iseb = cluster.layer() == PFLayer::ECAL_BARREL;
+    if (iseb){
+      EBSrFlagCollection::const_iterator srf = ebSrFlags->find(readOutUnitOf(static_cast<EBDetId>(cluster.seed())));      
+      clusFlag = srf->value();
```

Comment:
is it logically possible for the find to find nothing and srf to be invalid and impossible to dereference?
If so, there better be an appropriate protection.